### PR TITLE
4099 perm edit view

### PR DESCRIPTION
**Pull request by chiaberry**
# Description

Exit criteria: 
As an editor/publisher in a department

> I have permission to view the edit page for any page in my department
> I do not have permission to view the edit page for any page not in my department
> I have permission to save a draft of any page in my department
> I do not have permission to save a draft of any page not in my department

I added Page View Restrictions to the pages (https://www.laceyhenschel.com/blog/2018/8/1/search-history-adding-page-privacy-in-wagtail-programmatically) when pages are created. There is a migration to apply the restrictions to the pages that already exist in production. 

The BeforeEditHook checks to see if the user has permissions (aka is in the group that matches the PageViewPermissions' groups) and if they do not, there is a 404. Because the search results are filtered based on department, this will only happen if someone tries to access a page with a link directly. 

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 

https://joplin-pr-4099-perm-edit-view.herokuapp.com/
editor@austintexas.gov pw:e
member of aph

mod@austintexas.io pw: m
member of arr

a way I've been testing is to navigate to a page the user has permissions for and then change the url to a page above or below to test a different department's page. 

I also double checked that admins can still see all pages

# Checklist:
- [x] Request reviewers
- [x] Slack-out message for review
- [x] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: joplin/pages/base_page/migrations/0002_add_pageviewpermissions.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    JanisBasePage = apps.get_model('base_page.JanisBasePage') #todo: check this
+    PageViewRestriction = apps.get_model('wagtailcore.PageViewRestriction')
+
+    all_pages = JanisBasePage.objects.all()
+
+    # https://blog.etianen.com/blog/2013/06/08/django-querysets/
+    for page in all_pages.iterator():
+        pvr = PageViewRestriction.objects.create(
+            page=page,
+            restriction_type='groups',
```
Comment (chiaberry) [inline joplin/pages/base_page/migrations/0002_add_pageviewpermissions.py:18]: oddly enough, when I put `PageViewRestriction.GROUPS` just as it is on line 99 of new_page_from_modal above, the migrations errored. does anyone know why?

Comment (Niicck) [reply joplin/pages/base_page/migrations/0002_add_pageviewpermissions.py:18]: Models are weird when used inside migration files. It's not the same object as a real PageViewRestriction. In fact, it's a __fake__ PageViewRestriction:

Screen Shot 2020-06-04 at 10 14 03 AM

So not all attributes and methods are available. Why some things are restricted inside migration files, I don't know. I just trust that django knows what it's doing, and I copy techniques that other projects have used in their migration files to work around limitations.

File: joplin/templates/wagtailadmin/pages/listing/_page_title_explore.html [hunk 2]
```diff
@@ -15,7 +15,6 @@ <h2>
         {{ page.get_admin_display_title }}
     {% endif %}
 
-    {% include "wagtailadmin/pages/listing/_privacy_indicator.html" with page=page %}
```
Comment (chiaberry) [inline joplin/templates/wagtailadmin/pages/listing/_page_title_explore.html:18]: hides the privacy/eye symbol

Comment (Niicck) [reply joplin/templates/wagtailadmin/pages/listing/_page_title_explore.html:18]: No hay 👍

**Pull request merged**